### DISCUSSION TITLE
About channel types!

### DISCUSSION THREAD

**Discussion by sebastianjnuwu** [discordjs/discord.js#8025] [Q&A] [answered]
Hello. I could put only the text channels to appear without us having to make a Work around (in slash) 

```js
type: 'CHANNEL',
interaction.options.getChannel("bla");
```
But I just wanted the text channels ;-; without using If it could be like this:
```js
interaction.options.getTextChannel("bla");
```

**Comment by sebastianjnuwu** [comment]
Here is an example that could be a perfect command just click [here](https://github.com/sebastianjnuwu/KellyWorld/blob/0f4857936c2f30de1110beef2afe5771294a72a3/src/SlashCommands/SlashCommands/CommandSay.js) to see.....

**Comment by sebastianjnuwu** [comment]
Often we need to make a Work around and this is very annoying.....

**Comment by BONEE4** [comment]
This is a joke? lmao

**Reply by sebastianjnuwu** [reply]
> respostas

sorry

**Reply by kyranet** [reply]
Please be more respectful to other users, @BONEE4, I'll hide your comment this time, but I would appreciate if you take our [Code of Conduct](https://github.com/discordjs/discord.js/blob/main/.github/CODE_OF_CONDUCT.md) in mind next time.

**Comment by davipatricio** [comment, marked as answer]
If you are running `discordjs@dev`, you can create a command normally, and in the channel option, you can use the `channelTypes` property to tell what types of channels will be allowed.

Example:
```js
client.application.commands.create({
  name: 'test',
  description: 'A test command',
  options: [{
     type: ApplicationCommandOptionType.Channel,
     name: 'channel',
     description: 'Choose a channel.',
     channelTypes: [ChannelType.GuildText]
    }]
  })
```

**Reply by sebastianjnuwu** [reply]
Sorry, thanks for the help

**Reply by sebastianjnuwu** [reply]
> You can create a command normally, and in the channel option, you can use the `channelTypes` property to tell what types of channels will be allowed.
> 
> Example:
> 
> ```js
> client.application.commands.create({
>   name: 'test',
>   description: 'A test command',
>   options: [{
>      type: ApplicationCommandOptionType.Channel,
>      name: 'channel',
>      description: 'Choose a channel.',
>      channelTypes: [ChannelType.GuildText]
>     }]
>   })
> ```
> 
> And next time, don't use Brazilian slang, not everyone will understand. 😉

you can take a look [here](https://github.com/sebastianjnuwu/KellyWorld/blob/KellyWorld/src/SlashCommands/SlashCommands/CommandSay.js) returns me the error `ApplicationCommandOptionType.Channel` Is not definido

**Reply by davipatricio** [reply]
> > You can create a command normally, and in the channel option, you can use the `channelTypes` property to tell what types of channels will be allowed.
> > Example:
> > ```js
> > client.application.commands.create({
> >   name: 'test',
> >   description: 'A test command',
> >   options: [{
> >      type: ApplicationCommandOptionType.Channel,
> >      name: 'channel',
> >      description: 'Choose a channel.',
> >      channelTypes: [ChannelType.GuildText]
> >     }]
> >   })
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And next time, don't use Brazilian slang, not everyone will understand. 😉
> 
> you can take a look [here](https://github.com/sebastianjnuwu/KellyWorld/blob/KellyWorld/src/SlashCommands/SlashCommands/CommandSay.js) returns me the error `ApplicationCommandOptionType.Channel` Is not definido

In this case, you already imported the library, so you need to replace `ApplicationCommandOptionType` with `Discord.ApplicationCommandOptionType` and `ChannelType` with `Discord.ChannelType`

If you need further help, please, make questions at https://discord.gg/djs

**Comment by sebastianjnuwu** [comment]
Sorted out ,-,
Thanks davipatricio....
